pulp_smash/pulp_smash_internals.ipynb

```diff
@@ -0,0 +1,989 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Good to know\n",
+    "\n",
+    "In the simplest case, Pulp Smash's configuration file resides at\n",
+    "    ``~/.config/pulp_smash/settings.json``. However, there are several ways to\n",
+    "    alter this path. Pulp Smash obeys the `XDG Base Directory Specification`_.\n",
+    "    In addition, Pulp Smash responds to the ``PULP_SMASH_CONFIG_FILE``\n",
+    "    environment variable. This variable is a relative path, and it defaults to\n",
+    "    ``settings.json``.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running your first test"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "s\r\n",
+      "----------------------------------------------------------------------\r\n",
+      "Ran 0 tests in 0.000s\r\n",
+      "\r\n",
+      "OK (skipped=1)\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!python -m unittest pulp_smash.tests.pulp2.platform.api_v2.test_login"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Warning Explanation**\n",
+    "\n",
+    "If the configuration file states that HTTPS connections should not be verified, then InsecureRequestWarnings will be emitted. Smash is smart enough to suppress these warnings. But the unittest test runner overrides Smash and causes all warnings to be emitted anyway.\n",
+    "\n",
+    "\n",
+    "[Warning](https://github.com/PulpQE/pulp-smash/blob/2c5445972ff8c13cf32632f58b10b787ce2239cd/pulp_smash/tests/__init__.py#L43)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running all the tests "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All tests can be run by the following command. This action can take a while.\n",
+    "\n",
+    "``` bash\n",
+    "    python -m unittest discover pulp_smash.tests\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Installing Pulp-Smash in Development Mode\n",
+    "\n",
+    "In development mode changes to code are reflected in the working enviroment.\n",
+    "\n",
+    "\n",
+    "``` bash\n",
+    "git clone https://github.com/PulpQE/pulp-smash.git\n",
+    "cd pulp-smash\n",
+    "pip install --editable .[dev]\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Pulp-Fixtures\n",
+    "\n",
+    "Test environment and repeatability.\n",
+    "\n",
+    "https://github.com/PulpQE/pulp-fixtures\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "###  3 Main Components to be explained\n",
```

Comment:
```suggestion
    "### Main Components to be Explained\n",
```
It is bad form to use a number in a heading. If it is required, the number should be spelled out.

Since I don't believe "three" is adding any value here, I just removed it as the next sentence notes the component numbers

Also, for capitalization, techincal writing generally follows the MLA Handbook in which:

> In all three styles, you must capitalize nouns, pronouns, verbs, adjectives, and adverbs. In all three styles, do not capitalize articles, prepositions, or coordinating conjunctions.
> Visiting Beautiful Ruins (noun)
> As She Ran Away (pronoun)
> The Importance of Learning Fast (verb)
> The Poky Little Puppy (adjective)
> She Quietly Waits (adverb)

-- https://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html